torchgeo/samplers/single.py

```diff
@@ -156,6 +161,315 @@ def __len__(self) -> int:
         return self.length
 
 
+class TimeWindowGeoSampler(GeoSampler):
```

Comment:
Not sure what people think of the name, but I thought it's a bit more general. I have slightly extended the sampler to have the following functionalities:

1. `consecutive` argument. If true, then the target sequence follows the input sequence without any overlap in the time dimension. If false, then he target sequence will be the last prediction_length number of time steps from the input sequence end point. This would be intended to cover the use case of the Time Series task where the prediction is supposed to be the last input sequence step and not forecasting into the future.
2. `time_delta` argument: this specifies a time delta or gap between the input and the target sequence. Sometimes it is also called lead time I think where you want a time gap before the first prediction time step

So I think this sampler could support the following three use cases:
1. Time Series "Nowcasting": so for a time-series of images predict the latest target
2. Time Series "Forecasting": based on a historical sequence of images predict one or more future time steps
3. Time Series "Forecasting" with lead time: have a time gap between input and target sequence

I think it would not be so difficult to support another use case that gives more flexibility to the rolling window sampling we do over the time dimension. At the moment we are implicitly assuming as step size of 1 for the next sequence, but there could also be a shift argument that says how many time steps to move forward before generating the next input time sequence. For rolling window prediction (train on 3 days, predict next 24 hrs, shift 24 hrs).